pkg/agent/dozer/bcm/spec_vrf.go

```diff
@@ -447,7 +448,17 @@ var specVRFBGPNeighborEnforcer = &DefaultValueEnforcer[string, *dozer.SpecVRFBGP
 					EnableBfd: bfd,
 				},
 			},
-		}, nil
+		}
+
+		if value.UpdateSource != nil && *value.UpdateSource != "" {
+			bgpNeigh.Neighbor[name].Transport = &oc.OpenconfigNetworkInstance_NetworkInstances_NetworkInstance_Protocols_Protocol_Bgp_Neighbors_Neighbor_Transport{
+				Config: &oc.OpenconfigNetworkInstance_NetworkInstances_NetworkInstance_Protocols_Protocol_Bgp_Neighbors_Neighbor_Transport_Config{
+					LocalAddress: pointer.To(*value.UpdateSource),
```

Comment:
you don't need to deref and take a ref to it again here, just pass value.UpdateSource